### DISCUSSION TITLE
Multiple Aggregates on Same EventItem Container

### DISCUSSION THREAD

**Discussion by MattHartz** [IEvangelist/azure-cosmos-dotnet-repository#345] [General]
I have some aggregates that could in a standard SQL relationship might need to be transactionally bound. Does it make sense to potentially write events for both of these 2 aggregates using the same eventitem base class so that the writes would be transactional? I see there is a PersistAsync method that takes in a list of EventItem (is that expected to be used publically?). 

I would like to be able to do this through a middleware potentially and use some sort of change tracker. Does that make sense?

Thanks all!

**Comment by mumby0168** [comment]
Hi @MattHartz sorry for the delay, the event item is just an abstraction, in order for the two to be transactionally bound, they'd need to share the same partition key.

You can absolutely use that mentioned interface to write events in anyway you wish and also read them in such. I'd be keen to see an example of your use case + implementation.

Any more questions let me know.
